jdaviz/configs/imviz/helper.py

```diff
@@ -620,3 +620,7 @@ def link_image_data(app, link_type='pixels', wcs_fallback_scheme='pixels', wcs_u
                                              sender=app))
         # reset the progress spinner
         link_plugin.linking_in_progress = False
+
+    for viewer in app._viewer_store.values():
+        # viewer-state needs to know link type for reset_limits behavior
+        viewer.state.linked_by_wcs = link_type == 'wcs'
```

Comment:
Other than this, the rest looks fine at a glance. I don't grok all the limits stuff in core but minimal changes to Imviz tests means I don't worry.

Reply:
This is a... missing feature 😜  Working on adding support for new viewers to automatically adopt the correct value for `state.linked_by_wcs` so that their `reset_limits` works as expected based on the linking.

Reply:
the latest commit should handle this case (see recording).  Let me know if you like that solution or see any traps.

https://user-images.githubusercontent.com/877591/207409523-e69e694e-18d3-43c8-9e8c-a7ae87e4fb09.mov